### PR TITLE
Sign-in and Sign-up system

### PR DESCRIPTION
**Pull request by PeterTieu**

### CODE REVIEW

**Comment by ublefo** [comment]
There are a few things that needs to be addressed in this pull request:
1. You need provide reasonable description for the changes you have made.
2. One feature per pull request, 1k+ lines of changes makes this very difficult to review.
3. It seems like you are integrating kommunicate in the application, have this been discussed with the project lead? This is a 3rd party service that we don't control, you need to discuss this with the project lead before integrating external services that processes user data.

File: web-frontend/src/App.js [hunk 1]
```diff
@@ -96,6 +12,19 @@ import ForgotPassword from './routes/ForgotPassword/ForgotPassword';
 
 
 function App() {
+
+  useEffect(() => {
+    (function (d, m) {
+      var kommunicateSettings =
+        { "appId": "6ba050337f189922ca5b3c0e644a05e", "popupWidget": true, "automaticChatOpenOnNavigation": true };
```
Comment (ublefo) [inline web-frontend/src/App.js:19]: Credentials need to be stored in separate configuration files, not in the source code.

**Review by DARSHcverse — APPROVED** [review]
The code looks good because it's the same way my code works. So I can approve this.

**Review by DARSHcverse — APPROVED** [review]
All good! It works!

**Pull request merged**